src/hpa.c

```diff
@@ -499,10 +501,14 @@ hpa_try_hugify(tsdn_t *tsdn, hpa_shard_t *shard) {
 
 	malloc_mutex_unlock(tsdn, &shard->mtx);
 
-	shard->central->hooks.hugify(hpdata_addr_get(to_hugify), HUGEPAGE);
+	bool err = shard->central->hooks.hugify(hpdata_addr_get(to_hugify),
+	    HUGEPAGE, shard->opts.hugify_sync);
 
 	malloc_mutex_lock(tsdn, &shard->mtx);
 	shard->stats.nhugifies++;
+	if (err) {
+		shard->stats.nhugify_failures++;
```

Comment:
It's definitely a good idea to add a counter here. A question:
since `MADV_HUGEPAGE` is hugifying asynchronously, even it doesn't succeed in hugifying this range successfully ultimately, no errors will be returned here, is that correct? In production tests, with the sync turned off, the `nhugify_failures` should be 0?

Reply:
> It's definitely a good idea to add a counter here. A question:
since `MADV_HUGEPAGE` is hugifying asynchronously, even it doesn't succeed in hugifying this range successfully ultimately, no errors will be returned here, is that correct?

Yes, absolutely. Unless we are calling `madvise(..., MADV_HUGEPAGE)` with wrong arguments, but this will indicate a bug from our side.

> In production tests, with the sync turned off, the `nhugify_failures` should be 0?

Yes, that's correct.

Reply:
For better future tracking, maybe also add the above information to the comment? So that if users observe non-zero `nhugify_failures` with no `sync`, they can deduct that MADV_HUGEPAGE madvise is not executed properly? And when sync is enabled, this number should only reflect the successful rate of synchronous hugification.